File: public/dsa/two_pointers/remove_occurences/optimal/remove_occurences.cpp
#include <bits/stdc++.h>
using namespace std;

int removeOccurences(vector<int> A, int k) {
    int n = A.size();
    int j = 0;
    for (int i = 0; i < n; i++) {
        if (A[i] != k) {
            A[j] = A[i];
            j++;
        }
    }
    return j;
}

// tests-start

int main() {
    vector<int> A = {1, 4, 2, 6, 2, 6, 9, 4};
    int k = 4;
    cout << removeOccurences(A, k) << endl;
    return 0;
}

// tests-end

/*output
6
*/

File: public/dsa/two_pointers/remove_occurences/optimal/remove_occurences.py


from typing import List


class Solution:
    def removeOccurences(self, A: List[int], k: int) -> int:
        n = len(A)
        j = 0
        for i in range(n):
            if A[i] != k:
                A[j] = A[i]
                j += 1
        return j


# tests-start

A = [1, 4, 2, 6, 2, 6, 9, 4]
k = 4
sol = Solution()
print(sol.removeOccurences(A, k))

# tests-end

'''output
6
'''
